### DISCUSSION TITLE
Avoid first onChange

### DISCUSSION THREAD

**Discussion by singuerinc** [pmndrs/leva#439] [Q&A] [answered]
Hi, is there a way to avoid the onChange on the first render?

Let's suppose I want to change the url based on the value of `slug`.

```ts
useControls({
  slug: {
    options: ["post-1", "post-2", "post-3"],
    onChange: (slug) => {
      alert(slug); // post-1 is shown on page load
      router.push(slug) // will navigate on page load, not useful
    },
  },
});
```

I'm getting an infinite loop here. I know I could check the current url but I was wondering if is there a better solution.

**Comment by dbismut** [comment, marked as answer]
There's several arguments that onChange provides. The second or third one has an initial props that does what you want.

**Reply by singuerinc** [reply]
Thanks!

Solved it in this way:

```ts
useControls({
  slug: {
    options: ["post-1", "post-2", "post-3"],
    onChange: (slug, propName, options) => {
      if (!options.initial) {
          router.push(slug)
      }
    },
  },
});
```
